src/Serializer/ItemNormalizer.php

```diff
@@ -33,7 +33,16 @@ public function denormalize($data, $class, $format = null, array $context = [])
                 throw new InvalidArgumentException('Update is not allowed for this operation.');
             }
 
-            $context['object_to_populate'] = $this->iriConverter->getItemFromIri($data['id'], $context + ['fetch_data' => false]);
+            try {
+                $context['object_to_populate'] = $this->iriConverter->getItemFromIri($data['id'], $context + ['fetch_data' => false]);
+            } catch (InvalidArgumentException $e) {
+                foreach ($this->propertyNameCollectionFactory->create($context['resource_class'], $context) as $propertyName) {
+                    if (true === $this->propertyMetadataFactory->create($context['resource_class'], $propertyName)->isIdentifier()) {
```

Comment:
If we don't support yet composite identifiers, you can break after the first find. Or you can add support for composite identifiers :)